File: Lab8/graphQLService/src/main/kotlin/seg3x02/booksapigraphql/entity/Book.kt
package seg3x02.booksapigraphql.entity

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document(collection = "book")
data class Book(var bookNumber: Int,
                var category: String,
                var title: String,
                var cost: Float,
                var year: String?,
                var description: String?) {
    @Id
    var bookId: String = ""

    @Transient
    var authors: List<Author> = ArrayList()
}

File: Lab8/graphQLServiceClient/src/app/books/model/book.ts
export class Book {
  constructor(
    public bookId: number,
    public bookNumber: number,
    public category: string,
    public title: string,
    public cost: number,
    public authors?: Author[],
    public year?: number,
    public description?: string
  ) {}
}

export class Author {
  constructor(
    public bookNumber: number,
    public firstName: string,
    public lastName: string
  ){}
}
